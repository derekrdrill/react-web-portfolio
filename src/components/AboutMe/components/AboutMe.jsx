import React from 'react';
import styled, { createGlobalStyle } from 'styled-components';
import { Container, Row, Col } from 'react-bootstrap';
import { Typography, List, ListItem } from '@mui/material';

import { DarkLightModeContext } from '../../DarkLightMode/context/DarkLightModeContext';

import { ABOUT_ME_TEXT } from '../constants/ABOUT_ME_TEXT';
import bitmojiChillinWithBirds from '../../../assets/bitmoji_chillinWithBirds1.png';

export const AboutMe = ({ id }) => {
  // test change

  const { darkMode } = React.useContext(DarkLightModeContext);

  return (
    <>
      <PageBodyStyle darkMode={darkMode} />
      <AboutMeContainer darkMode={darkMode} id={id} fluid>
        <Row>
          <Col className='py-3'>
            <AboutMeTitle className='text-center' component='h1' darkMode={darkMode} variant='h3'>
              ABOUT ME
            </AboutMeTitle>
          </Col>
        </Row>
        <Row>
          <AboutMeTextContainer
            darkMode={darkMode}
            xs={{ span: 10, order: 'last', offset: 1 }}
            lg={{ span: 5, order: 'first', offset: 1 }}
          >
            <AboutMeText darkMode={darkMode} variant='subtitle1' component='p'>
              {ABOUT_ME_TEXT[1].text}
            </AboutMeText>
            <AboutMeText darkMode={darkMode} variant='subtitle1' component='p'>
              {ABOUT_ME_TEXT[2].text}
            </AboutMeText>
            <List style={{ color: 'white' }}>
              <AboutMeListItem darkMode={darkMode}>{ABOUT_ME_TEXT[3].text}</AboutMeListItem>
              <AboutMeListItem darkMode={darkMode}>{ABOUT_ME_TEXT[4].text}</AboutMeListItem>
              <AboutMeListItem darkMode={darkMode}>{ABOUT_ME_TEXT[5].text}</AboutMeListItem>
              <AboutMeListItem darkMode={darkMode}>{ABOUT_ME_TEXT[6].text}</AboutMeListItem>
            </List>
            <AboutMeText darkMode={darkMode} variant='subtitle1' component='p'>
              {ABOUT_ME_TEXT[7].text}
            </AboutMeText>
            <AboutMeText darkMode={darkMode} variant='subtitle1' component='p'>
              {ABOUT_ME_TEXT[8].text}
            </AboutMeText>
            <AboutMeText darkMode={darkMode} variant='subtitle1' component='p'>
              {ABOUT_ME_TEXT[9].text}
            </AboutMeText>
          </AboutMeTextContainer>
          <AboutMeBitmojiContainer xs={12} lg={6}>
            <img src={bitmojiChillinWithBirds} alt='Chillin with birds' />
          </AboutMeBitmojiContainer>
        </Row>
      </AboutMeContainer>
    </>
  );
};

const PageBodyStyle = createGlobalStyle(({ darkMode }) => ({
  body: {
    backgroundColor: darkMode ? '#030200' : 'whitesmoke',
  },
}));

export const AboutMeContainer = styled(Container)(({ darkMode }) => ({
  padding: '30px 50px 200px 50px',
  borderTop: '2px solid transparent',
  borderImage: darkMode
    ? 'linear-gradient(to right, rgba(248, 184, 255, 1), skyblue, gainsboro)'
    : 'linear-gradient(to right, rgba(176, 52, 197, 1), rgba(86, 206, 210, 1))',
  borderImageSlice: 1,
}));

export const AboutMeTitle = styled(Typography)(({ darkMode }) => ({
  fontFamily: 'Shizuru',
  fontWeight: darkMode ? 'normal' : 'bold',
  color: darkMode ? 'gainsboro' : 'darkslategray',
}));

export const AboutMeTextContainer = styled(Col)(({ darkMode }) => ({
  padding: 40,
  height: 550,
  overflowY: 'auto',
  border: '1px dashed #383838',
  borderRadius: 2,
  backgroundColor: darkMode ? '#303030' : '#eee5c1',
}));

export const AboutMeBitmojiContainer = styled(Col)({
  textAlign: 'center',
  paddingTop: 75,
  paddingBottom: 20,
});

export const AboutMeText = styled(Typography)(({ darkMode }) => ({
  fontFamily: 'Kufam',
  color: darkMode ? 'gainsboro' : '#333333',
  marginBottom: 15,
}));

export const AboutMeListItem = styled(ListItem)(({ darkMode }) => ({
  fontFamily: 'Kufam',
  color: darkMode ? 'gainsboro' : '#333333',
}));
