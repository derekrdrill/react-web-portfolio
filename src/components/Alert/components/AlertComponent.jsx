import React from 'react';
import styled, { css, keyframes } from 'styled-components';
import { Alert, AlertTitle } from '@mui/material';
import { AlertContext } from '../context/AlertContext';
import { DarkLightModeContext } from '../../DarkLightMode/context/DarkLightModeContext';

export const AlertComponent = () => {
  const { alert, fadeOut } = React.useContext(AlertContext);
  const { darkMode } = React.useContext(DarkLightModeContext);

  return (
    alert && (
      <StyledAlert darkMode={darkMode} severity={alert.type} $fadeOut={fadeOut}>
        <AlertTitle>{alert.title}</AlertTitle>
        {alert.msg}
      </StyledAlert>
    )
  );
};

const fadeOut = keyframes({
  '0%': {
    opacity: 1,
  },
  '100%': {
    opacity: 0,
  },
});

export const StyledAlert = styled(Alert)`
  animation-name: ${props =>
    props.$fadeOut
      ? css`
          ${fadeOut}
        `
      : 'none'};
  animation-duration: 500ms;
  animation-direction: linear;
`;
