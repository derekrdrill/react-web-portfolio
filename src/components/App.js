import React from 'react';

const AboutMe = React.lazy(() => import('./AboutMe/components/AboutMe'));
const ConnectWithMe = React.lazy(() => import('./ConnectWithMe/components/ConnectWithMe'));
const Homepage = React.lazy(() => import('./Homepage/Homepage'));
const Projects = React.lazy(() => import('./Projects/components/Projects'));

const App = () => (
  <div className='App'>
    <Homepage id='home' />
    <AboutMe id='about-me' />
    <Projects id='projects' />
    <ConnectWithMe id='connect-with-me' />
  </div>
);

export default App;
