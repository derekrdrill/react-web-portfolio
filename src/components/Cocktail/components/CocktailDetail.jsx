import React from 'react';
import styled from 'styled-components';
import { Grid, Typography } from '@mui/material';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faPlayCircle } from '@fortawesome/fontawesome-free-solid';

import { CocktailContext } from '../../Cocktail/context/CocktailContext';
import { DarkLightModeContext } from '../../DarkLightMode/context/DarkLightModeContext';

import { LoaderSpinner } from '../../LoaderSpinner/LoaderSpinner';

const CocktailDetail = () => {
  const { searchResults, searchResultsLength, loading } = React.useContext(CocktailContext);
  const { darkMode } = React.useContext(DarkLightModeContext);

  return (
    <>
      {loading && <LoaderSpinner open />}
      {!loading && searchResults && searchResults.length > 0 && (
        <CocktailDetailContainer container rowSpacing={2}>
          <Grid item xs={1} md={2} />
          <Grid item xs={10}>
            <Typography variant='h5'>
              {`${searchResultsLength} ${searchResultsLength === 1 ? 'drink' : 'drinks'} listed`}
            </Typography>
          </Grid>
          <Grid item xs={12} />
          <Grid item xs={12} />
          {searchResults.map((searchResult, searchResultKey) => (
            <>
              <CocktailResultsContainer key={searchResultKey} container rowSpacing={4}>
                <Grid item xs={1} md={2} />
                <Grid item xs={11} md={3}>
                  <CocktailImage src={searchResult.strDrinkThumb} />
                </Grid>
                <Grid item xs={1} sm={1} />
                <Grid item xs={10} md={5}>
                  <Typography variant='h3'>{searchResult.strDrink}</Typography>
                  <br />
                  <Grid container spacing={4}>
                    <Grid item xs={12} sm={6}>
                      <Typography variant='h5'>Ingredients</Typography>
                      {Object.keys(searchResult)
                        .filter(result => result.includes('strIngredient'))
                        .map(
                          (ingredient, ingredientKey) =>
                            searchResult[ingredient] && (
                              <Typography
                                key={`${searchResult.idDrink}${ingredientKey}`}
                                variant='subtitle2'
                              >
                                {`${searchResult[`strMeasure${ingredientKey + 1}`] ?? ''} ${
                                  searchResult[ingredient]
                                }`}
                              </Typography>
                            ),
                        )}
                    </Grid>
                    <Grid item xs={12} sm={6}>
                      <Typography variant='h5'>Glass</Typography>
                      <Typography variant='subtitle2'>{searchResult.strGlass}</Typography>
                    </Grid>
                  </Grid>
                  <br />
                  <Typography variant='h5'>Instructions</Typography>
                  <Typography variant='body2'>{searchResult.strInstructions}</Typography>
                  {searchResult.youtubeData && (
                    <CocktailVideoTutorialLinkContainer item xs={12}>
                      <CocktailVideoTutorialLink
                        href={searchResult.youtubeData.url}
                        target='_blank'
                        rel='noreferrer'
                      >
                        <CocktailVideoTutorial
                          className='youtube-img'
                          src={searchResult.youtubeData.bestThumbnail.url}
                        />
                        <CocktailVideoTutorialPlayButton icon={faPlayCircle} />
                        <CocktailVideoTutorialTitleContainer className='youtube-title'>
                          <CocktailVideoTutorialText variant='body1'>
                            {searchResult.youtubeData.title}
                          </CocktailVideoTutorialText>
                          <CocktailVideoTutorialText variant='subtitle2'>
                            {searchResult.youtubeData.duration}
                          </CocktailVideoTutorialText>
                        </CocktailVideoTutorialTitleContainer>
                      </CocktailVideoTutorialLink>
                    </CocktailVideoTutorialLinkContainer>
                  )}
                </Grid>
                <Grid xs={1} />
                <Grid xs={1} />
                <CocktailResultsDivider
                  item
                  xs={10}
                  darkMode={darkMode}
                  lastItem={searchResultKey + 1 === searchResultsLength}
                />
                <Grid xs={1} />
              </CocktailResultsContainer>
            </>
          ))}
        </CocktailDetailContainer>
      )}
    </>
  );
};

export default CocktailDetail;

export const CocktailDetailContainer = styled(Grid)({
  marginBottom: 150,
  position: 'relative',
  top: 100,
});

export const CocktailImage = styled.img({
  borderRadius: 30,
  height: 275,
  width: 320,
});

export const CocktailResultsContainer = styled(Grid)({
  padding: '40px 0',
});

export const CocktailResultsDivider = styled(Grid)(({ darkMode, lastItem }) => ({
  borderBottom: !lastItem && `solid 1px ${darkMode ? 'gainsboro' : 'grey'}`,
  margin: '0 30px',
  padding: '50px 0',
}));

export const CocktailVideoTutorial = styled.img({
  height: 215,
  width: 315,
});

export const CocktailVideoTutorialLinkContainer = styled(Grid)({
  ':hover': {
    '.youtube-img': {
      opacity: 0.4,
    },
    '.youtube-title': {
      visibility: 'visible',
    },
  },
  height: 215,
  transform: 'translateY(20px)',
  width: 315,
});

export const CocktailVideoTutorialLink = styled.a({
  cursor: 'pointer',
  textDecoration: 'none',
  zIndex: 2,
});

export const CocktailVideoTutorialText = styled(Typography)({
  color: 'white',
  fontWeigt: 'bold',
  zIndex: 0,
});

export const CocktailVideoTutorialTitleContainer = styled.div({
  maxWidth: 300,
  transform: 'translate(10px, -250px)',
  visibility: 'hidden',
});

export const CocktailVideoTutorialPlayButton = styled(FontAwesomeIcon)({
  color: 'red',
  height: 30,
  transform: 'translate(145px, -130px)',
  width: 30,
});
