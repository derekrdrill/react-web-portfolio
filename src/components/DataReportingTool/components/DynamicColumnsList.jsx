import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { Grid, IconButton, List, ListItem, Typography } from '@mui/material';
import AddCircleIcon from '@mui/icons-material/AddCircle';
import RemoveCircleIcon from '@mui/icons-material/RemoveCircle';

export const getColumnName = columnName => columnName ?? '';
export const getIconButtonColor = initialList => (initialList ? 'success' : 'error');
export const getIconButtonChildren = initialList => (initialList ? <AddCircleIcon /> : <RemoveCircleIcon />);
export const getFirstGridItemOrderLG = initialList => (initialList ? 1 : 2);
export const getSecondGridItemOrderLG = initialList => (initialList ? 2 : 1);
export const getFirstGridContainerJustifyLG = initialList => (initialList ? 'flex-start' : 'flex-end');
export const getSecondGridContainerJustifyLG = initialList => (initialList ? 'flex-end' : 'flex-start');

export const DynamicColumnsList = ({ columns, initialList = false, buttonClick }) => (
  <ColumnsListContainer>
    <ColumnsList>
      {columns.map(column => (
        <ColumnsListItem key={column.id} draggable='true'>
          <Grid container>
            <Grid item xs={10} order={{ xs: 1, lg: getFirstGridItemOrderLG(initialList) }}>
              <Grid
                container
                justifyContent={{
                  xs: 'flex-start',
                  lg: getFirstGridContainerJustifyLG(initialList),
                }}
              >
                <Typography variant='subtitle1' component='h3'>
                  {getColumnName(column.columnName)}
                </Typography>
              </Grid>
            </Grid>
            <Grid item xs={2} order={{ xs: 2, lg: getSecondGridItemOrderLG(initialList) }}>
              <Grid
                container
                justifyContent={{
                  xs: 'flex-end',
                  lg: getSecondGridContainerJustifyLG(initialList),
                }}
              >
                <IconButton
                  id={column.id}
                  color={getIconButtonColor(initialList)}
                  onClick={
                    /* istanbul ignore next */
                    e => buttonClick(e, initialList)
                  }
                  variant='contained'
                >
                  {getIconButtonChildren(initialList)}
                </IconButton>
              </Grid>
            </Grid>
          </Grid>
        </ColumnsListItem>
      ))}
    </ColumnsList>
  </ColumnsListContainer>
);

DynamicColumnsList.propTypes = {};

const ColumnsListContainer = styled.div({
  height: 250,
  overflowY: 'auto',
  overflowX: 'none',
  border: '1px solid grey',
  borderRadius: 5,
  boxShadow: '0 0 3px darkslategray',
  backgroundColor: '#EDE9E1',
});

const ColumnsList = styled(List)({
  padding: 0,
});

const ColumnsListItem = styled(ListItem)({
  backgroundColor: 'darkgrey',
  borderBottom: '1px dashed grey',
  cursor: 'all-scroll',
  padding: 10,
});
