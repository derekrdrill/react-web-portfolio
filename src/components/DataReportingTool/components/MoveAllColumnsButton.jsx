import React from 'react';
import PropTypes from 'prop-types';
import { Button } from '@mui/material';

export const MoveAllColumnsButton = ({ children, endIcon, onClick }) => (
  <Button children={children} color='secondary' endIcon={endIcon} fullWidth onClick={onClick} variant='contained' />
);
