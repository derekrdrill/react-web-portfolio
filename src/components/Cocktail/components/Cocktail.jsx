import React from 'react';
import PropTypes from 'prop-types';
import styled, { createGlobalStyle } from 'styled-components';

import { AlertProvider } from '../../Alert/context/AlertContext';
import { DarkLightModeContext } from '../../DarkLightMode/context/DarkLightModeContext';

import CocktailDetail from './CocktailDetail';
import CocktailSearch from './CocktailSearch';

const Cocktail = () => {
  const { darkMode } = React.useContext(DarkLightModeContext);

  return (
    <AlertProvider>
      <PageStyle darkMode={darkMode} />
      <CocktailSearch />
      <CocktailDetail />
    </AlertProvider>
  );
};

Cocktail.propTypes = {};

export default Cocktail;

const PageStyle = createGlobalStyle(({ darkMode }) => ({
  body: {
    backgroundColor: darkMode ? '#1B0008' : '#FBAF93',
  },
  'h1, h2, h3, h4, h5, h6, p': {
    fontFamily: 'Trebuchet MS, sans-serif',
    fontWeight: 'bold',
    color: darkMode ? 'beige' : '#464646',
  },
}));
