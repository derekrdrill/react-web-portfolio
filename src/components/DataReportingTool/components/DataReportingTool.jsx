import React, { useState, useEffect } from 'react';
import styled, { createGlobalStyle } from 'styled-components';
import { Select, InputLabel, FormControl, Typography, MenuItem, Grid, Button } from '@mui/material';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faLongArrowAltDown, faLongArrowAltUp } from '@fortawesome/fontawesome-free-solid';

import { DynamicColumnsList } from '../components/DynamicColumnsList';
import { MoveAllColumnsButton } from '../components/MoveAllColumnsButton';

export const DataReportingTool = () => {
  const [dataSet, setDataSet] = useState('Select One');
  const [collections, setCollections] = useState([]);
  const [collectionKeys, setCollectionKeys] = useState([]);
  const [availableColumns, setAvailableColumns] = useState([]);
  const [associatedColumns, setAssociatedColumns] = useState([]);

  const handleGetAllCollections = () => {
    fetch('http://localhost:3001/listCollections')
      .then(response => response.json())
      .then(json => setCollections(json));
  };

  const handleGetCollectionKeys = React.useCallback(() => {
    fetch(`http://localhost:3001/listCollectionKeys/${dataSet}`)
      .then(response => response.json())
      .then(json => setCollectionKeys(json));
  }, [dataSet]);

  const handleColumnState = (e, initialList) => {
    let columnID = Number(e.currentTarget.id);
    let columns = initialList ? availableColumns : associatedColumns;
    let columnObject = columns.find(column => column.id === columnID);

    if (initialList) {
      setAvailableColumns(availableColumns.filter(column => column.id !== columnID));
      setAssociatedColumns([...associatedColumns, columnObject]);
    } else {
      setAssociatedColumns(associatedColumns.filter(column => column.id !== columnID));
      setAvailableColumns([...availableColumns, columnObject]);
    }
  };

  const handleAssociateAll = () => {
    if (availableColumns.length > 0) {
      setAssociatedColumns(associatedColumns.concat(availableColumns));
      setAvailableColumns([]);
    }
  };

  const handleDeassociateAll = () => {
    if (associatedColumns.length > 0) {
      setAvailableColumns(availableColumns.concat(associatedColumns));
      setAssociatedColumns([]);
    }
  };

  useEffect(() => handleGetAllCollections(), []);
  useEffect(() => handleGetCollectionKeys(), [dataSet, handleGetCollectionKeys]);
  useEffect(() => setAvailableColumns(collectionKeys), [collectionKeys]);

  return (
    <div>
      <PageBodyStyle />
      <DataSetSelectContainer container>
        <Grid item xs={3}>
          <FormControl fullWidth>
            <InputLabel>
              <Typography variant='subtitle1' component='label'>
                Select dataset
              </Typography>
            </InputLabel>
            <DataSetSelect
              label={dataSet}
              variant='standard'
              onChange={e => setDataSet(e.target.value)}
              value={dataSet ?? ''}
            >
              {collections.map(collection => (
                <MenuItem key={collection.name} value={collection.name}>
                  {collection.name}
                </MenuItem>
              ))}
              {/* <MenuItem value='leadData'>Lead Data</MenuItem> */}
            </DataSetSelect>
          </FormControl>
        </Grid>
      </DataSetSelectContainer>
      <ColumnsSelectContainer container>
        <Grid item xs={12} lg={5}>
          <Typography variant='h5' component='h2'>
            Available Columns
          </Typography>
          <DynamicColumnsList
            columns={availableColumns}
            buttonClick={handleColumnState}
            initialList
          />
        </Grid>
        <MoveAllColumnsButtonContainer item xs={12} lg={2}>
          <Grid container justifyContent='center' spacing={{ xs: 1, lg: 2 }}>
            <Grid item xs={6} lg={7} order={{ xs: 2, lg: 1 }}>
              <Grid item display={{ xs: 'none', lg: 'block' }}>
                <MoveAllColumnsButton onClick={handleAssociateAll}>{`>`}</MoveAllColumnsButton>
              </Grid>
              <Grid item display={{ xs: 'block', lg: 'none' }}>
                <MoveAllColumnsButton
                  endIcon={<FontAwesomeIcon icon={faLongArrowAltDown} />}
                  onClick={handleAssociateAll}
                >
                  {`ASSOCIATE ALL`}
                </MoveAllColumnsButton>
              </Grid>
            </Grid>
            <Grid item xs={6} lg={7} order={{ xs: 1, lg: 2 }}>
              <Grid item display={{ xs: 'none', lg: 'block' }}>
                <MoveAllColumnsButton onClick={handleDeassociateAll}>{`<`}</MoveAllColumnsButton>
              </Grid>
              <Grid item display={{ xs: 'block', lg: 'none' }}>
                <MoveAllColumnsButton
                  endIcon={<FontAwesomeIcon icon={faLongArrowAltUp} />}
                  onClick={handleDeassociateAll}
                >
                  {`DEASSOCIATE ALL`}
                </MoveAllColumnsButton>
              </Grid>
            </Grid>
          </Grid>
        </MoveAllColumnsButtonContainer>
        <Grid item xs={12} lg={5}>
          <Typography variant='h5' component='h2'>
            Associated Columns
          </Typography>
          <DynamicColumnsList columns={associatedColumns} buttonClick={handleColumnState} />
        </Grid>
      </ColumnsSelectContainer>
      <hr />
      <Grid container justifyContent='center'>
        <Grid item xs={12} lg={4}>
          <Button color='info' variant='contained' fullWidth>{`Create Report`}</Button>
        </Grid>
      </Grid>
    </div>
  );
};

const PageBodyStyle = createGlobalStyle({
  body: {
    backgroundColor: '#228881',
  },
});

const DataSetSelectContainer = styled(Grid)({
  backgroundColor: 'grey',
  // position: '-webkit-sticky',
  position: 'sticky',
  top: 80,
  padding: '10px 0 15px 0',
  zIndex: 100,
});

const DataSetSelect = styled(Select)({
  backgroundColor: 'gainsboro',
  paddingLeft: 10,
  borderRadius: 5,
  marginLeft: 10,
});

const ColumnsSelectContainer = styled(Grid)({
  padding: 20,
});

// const ColumnsListContainer = styled.div({
//   height: 250,
//   overflowY: 'auto',
//   overflowX: 'none',
//   border: '1px solid grey',
//   borderRadius: 5,
//   boxShadow: '0 0 3px darkslategray',
//   backgroundColor: '#EDE9E1',
// });

// const ColumnsList = styled(List)({
//   padding: 0,
// });

// const ColumnsListItem = styled(ListItem)({
//   backgroundColor: 'darkgrey',
//   borderBottom: '1px dashed grey',
//   cursor: 'all-scroll',
//   padding: 10,
// });

const MoveAllColumnsButtonContainer = styled(Grid)({
  margin: '7% 0 4% 0',
});
