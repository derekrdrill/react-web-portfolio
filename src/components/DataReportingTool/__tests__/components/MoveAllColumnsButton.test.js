import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import { MoveAllColumnsButton } from '../../components/MoveAllColumnsButton';

const renderer = new ShallowRenderer();

describe('MoveAllColumnsButton with me tests', () => {
  it('renders correctly', () => {
    renderer.render(<MoveAllColumnsButton children='TEST' onClick={jest.fn()} />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
