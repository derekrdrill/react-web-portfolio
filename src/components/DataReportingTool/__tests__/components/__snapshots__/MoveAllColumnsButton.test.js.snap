// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoveAllColumnsButton with me tests renders correctly 1`] = `
<ForwardRef(Button)
  color="secondary"
  fullWidth={true}
  onClick={[MockFunction]}
  variant="contained"
>
  TEST
</ForwardRef(Button)>
`;
